src/hotspot/cpu/x86/x86_64.ad

```diff
@@ -7443,14 +7498,70 @@ instruct decI_mem(memory dst, immI_M1 src, rFlagsReg cr)
   ins_pipe(ialu_mem_imm);
 %}
 
-instruct leaI_rReg_immI(rRegI dst, rRegI src0, immI src1)
+instruct leaI_rReg_immI2_immI(rRegI dst, rRegI index, immI2 scale, immI disp)
 %{
-  match(Set dst (AddI src0 src1));
+  predicate(VM_Version::supports_fast_2op_lea());
```

Comment:
We actually have three operands (index, scale and dis) for this rule.
Shouldn't this predicate be `VM_Version::supports_fast_3op_lea()`?

Please make it clear what do you mean by `supports_fast_3op_lea`.
I guess you mean 3 operands including `base, index and disp`, right?

Reply:
Yes, I have clarified this in the comment of `VM_Version::supports_fast_3op_lea()`